components/profile-controller/README.md

```diff
@@ -0,0 +1,72 @@
+# Profile CRD
+
+Kubeflow Profile CRD is designed to solve access management within multi-user kubernetes cluster.
+
+Profile access management provide namespace level isolation based on:
+
+- k8s rbac access control
+- Istio rbac access control
+
+**Resources managed by profile CRD:**
+
+Each profile CRD will manage one namespace (with same name as profile CRD), and will have one owner.
+Specifically, each profile CRD will manage following resources:
+
+- Namespace
+- K8s rbac Role & Rolebinding to grant profile owner access to above namespace via k8s API (kubectl)
+- Istio namespace-scoped gateway & rbac permission to grant profile owner access to above namespace via Istio (browser)
+- Setup namespace-scoped service-accounts `editor` and `viewer` to be used by user-created pods in 
+- resource quota management (coming)
+
+## Supported platforms and prerequisites:
+
+**GCP**
+- All users should have IAM permission `Kubernetes Engine Cluster Viewer`
+  - This is needed in order to get cluster access by `gcloud container clusters get-credentials`
+- kubeflow cluster with version v0.6
+
+## Manage access control and resources
+
+### manual access management by admin
+
+Cluster admin will manage access management for cluster users:
+
+**To create and reserve namespace `ns1` for user `abc@def.com`**
+- create profile:
+```
+apiVersion: kubeflow.org/v1alpha1
+kind: Profile
+metadata:
+  name: ns1
+spec:
+  owner:
+    kind: User
+    name: abc@def.com
+```
+
+**To revoke access to namespace `ns1` from user `abc@def.com` and delete namespace `ns1`**
+- delete profile ns1:
+```
+kubectl delete profile ns1
+```
+
+You should have kubeflow components deployed inside your k8s cluster. Generated ksonnet application is store in [app-dir](./bootstrapper.yaml#L65).
+Exec into pod ```kubeflow-bootstrapper-0``` in namespace ```kubeflow-admin``` if you need to edit your ksonnet app.
+
+The default components are defined in [default.yaml](config/default.yaml), user can customize which components to deploy by
+pointing ```--config``` args in [bootstrapper.yaml](./bootstrapper.yaml) to their own config (eg. a configmap in k8s cluster)
+
+This bootstrapper example [config](config/gcp_prototype.yaml) can help explain how config customization works.
+
+### Dev Instruction
+
+##### How to generate Istio rbac CRD types
+
+- We use kube builder https://book.kubebuilder.io/quick_start.html
+```
+kubebuilder init --domain istio.io --license apache2 --owner "The Kubernetes Authors"
+kubebuilder create api --group rbac --version v1alpha1 --kind ServiceRole
+kubebuilder create api --group rbac --version v1alpha1 --kind ServiceRoleBinding
+```
+- Then copy ServiceRole / ServiceRoleBinding schema from Istio release version https://github.com/istio/istio/releases/tag/1.1.6 to `pkg/apis/istiorbac/v1alpha1/*_types.go`
```

Comment:
Was this a one time activity?

Reply:
This instruction indicate how the Istio code are generated.
Should be updated no more than twice a year. (usually when Istio has breaking API change)